### PR TITLE
porting hstore contrib module docs from postgres 8.4

### PR DESCRIPTION
**Pull request by dyozie**

### CODE REVIEW

**Comment by mkiyama** [comment]
Should hstore be added to the GPDB data types?

**Comment by dyozie** [comment]
@mkiyama - The current datatypes doc prefaces the list as "native" data types, so I'm not sure we'd want to include something provided in a contrib module there.  Might be nice to refer out to hstore though.

**Comment by mkiyama** [comment]
Just a thought...

This is in the description of data types.
"This reference shows all of the built-in data types"

hstore will be "built in". As part of the description, there could be a link to the contrib module topic and a note saying it is a postgres contrib module.

**Comment by Eulerizeit** [comment]
I posted this in the wrong place before. Not sure quite what happened.

Mel,

It will be built in only by a loose definition. In order to use it you will
need to run an hstore.sql file

-- Rob

**Comment by mkiyama** [comment]
@Eulerizeit @dyozie 
That was my mistake. I posted my "built-in" comment in the wrong PR. sorry.
Rob, thanks for the clarification about hstore

David,  I didn't see the part about needing to run an hstore.sql file in the PR.

**Comment by mkiyama** [comment]
@Eulerizeit @ivannovick 
Is hstore supported by Pivotal in the "enterprise" build, or is this a oss only addition.

**Comment by Eulerizeit** [comment]
the way to install it would be to run the hstore.sql . "\i <install
dir>/share/hstore.sql"

**Comment by Eulerizeit** [comment]
Enterprise build. To my knowledge there is nothing yet in the OSS product
that isn't in the enterprise.

I can't wait for that to happen.

-- Rob

**Comment by dyozie** [comment]
Rob:  I thought this might be one of the first modules to support CREATE
EXTENSION.  Are you sure the script needs to be run manually?

**Comment by Eulerizeit** [comment]
@dyozie I honestly didn't look at the PR so I don't know. 

@hlinnaka does this use the extension framework or do users still need to run the hstore.sql file?

**Comment by hlinnaka** [comment]
On 08/29/2017 01:25 AM, Robert Eckhardt wrote:
> @hlinnaka does this use the extension framework or do users still need to run the hstore.sql file?

Does not use the extension framework, you need to run the hstore.sql file.

- Heikki

File: gpdb-doc/dita/utility_guide/hstore.xml [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+       <entry><codeph>"a"=>"1"</codeph></entry>
+      </row>
+     </tbody>
+    </tgroup>
+   </table>
+  </section>
+  <section>
+   <title>Indexes</title>
+   <p><codeph>hstore</codeph> has index support for <codeph>@&gt;</codeph> and <codeph>?</codeph>
+    operators. You can use either GiST or GIN index types. For example:</p>
```
Comment (hlinnaka) [inline gpdb-doc/dita/utility_guide/hstore.xml]: GPDB doesn't support GIN indexes, so that won't work. GiST indexes do work.

**Pull request merged**